vtex/faststore
# Feature/generic ext provider

## What's the purpose of this pull request?

Previously we were hardcoding the auth providers and repeating code for redirect ones (such as google and facebook). The idea is to actually implement a bigger part of the OAuth flow, retrieve providers and then mount the ones that are actually available on the store.

## How it works? 

It creates a list of default components and filters that based on the providers registered on the account. 

## How to test it?

On boticario and marinbrasil it should display regularly the user/pwd, token retrieval, google and facebook login options and on carrefour it should automatically redirect to their login provider.

## References

All the knowledge acquired to make this PR is contained [in this thread](https://vtex.slack.com/archives/G01PE5NSST1/p1615819290108600)
